Source/Tools/FEXLogServer/Main.cpp

```diff
@@ -0,0 +1,268 @@
+    std::erase_if(PIDs, [FD](auto& Entry) {
+      Entry.second.FDs.erase(FD);
+      return Entry.second.FDs.empty();
+    });
+    FEX::GUI::HadUpdate();
+  }
+
+  bool LoglineFiller(void *data, int idx, const char** out_text) {
+    std::unique_lock lk{LogMutex};
+
+    if (idx < LogLines.size()) {
+      auto &Log = LogLines.at(idx);
+      *out_text = Log.Log.c_str();
+
+      return true;
+
+    }
+    return false;
+  }
+
+  void DrawUI() {
+    ImGuiIO& io = ImGui::GetIO();
+    auto current_time = std::chrono::high_resolution_clock::now();
+    auto Diff = std::chrono::duration_cast<std::chrono::milliseconds>(current_time - GlobalTime);
+    io.DeltaTime = Diff.count() > 0 ? Diff.count() : 1.0f/60.0f;
+    GlobalTime = current_time;
+
+    ImGui::NewFrame();
+
+    // We are using the ImGuiWindowFlags_NoDocking flag to make the parent window not dockable into,
+    // because it would be confusing to have two docking targets within each others.
+    ImGuiWindowFlags window_flags = ImGuiWindowFlags_MenuBar | ImGuiWindowFlags_NoDocking;
+    ImGuiViewport* viewport = ImGui::GetMainViewport();
+    ImGui::SetNextWindowPos(viewport->Pos);
+    ImGui::SetNextWindowSize(viewport->Size);
+    ImGui::SetNextWindowViewport(viewport->ID);
+
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding, 0.0f);
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowBorderSize, 0.0f);
+    window_flags |= ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove;
+    window_flags |= ImGuiWindowFlags_NoBringToFrontOnFocus | ImGuiWindowFlags_NoNavFocus;
+
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2(0.0f, 0.0f));
+    ImGui::Begin("DockSpace", nullptr, window_flags);
+    ImGui::PopStyleVar(3);
+
+    ImGui::DockSpace(ImGui::GetID("DockSpace"));
+
+    if (ImGui::BeginMenuBar()) {
+      // XXX: Nothing here yet
+      ImGui::EndMenuBar();
+    }
+
+    {
+      if (ImGui::Begin("#Config")) {
+        if (ImGui::RadioButton("Info", LevelFilter == LogMan::DebugLevels::INFO)) {
+          LevelFilter = LogMan::DebugLevels::INFO;
+        }
+        if (ImGui::RadioButton("Debug", LevelFilter == LogMan::DebugLevels::DEBUG)) {
+          LevelFilter = LogMan::DebugLevels::DEBUG;
+        }
+        if (ImGui::RadioButton("Error", LevelFilter == LogMan::DebugLevels::ERROR)) {
+          LevelFilter = LogMan::DebugLevels::ERROR;
+        }
+        if (ImGui::RadioButton("Assert", LevelFilter == LogMan::DebugLevels::ASSERT)) {
+          LevelFilter = LogMan::DebugLevels::ASSERT;
+        }
+        if (ImGui::RadioButton("None", LevelFilter == LogMan::DebugLevels::NONE)) {
+          LevelFilter = LogMan::DebugLevels::NONE;
+        }
+
+        ImGui::Checkbox("Print to stderr", &AlsoCLI);
+
+        if (ImGui::TreeNodeEx("PIDs", ImGuiTreeNodeFlags_DefaultOpen)) {
+          std::unique_lock lk{PIDMappingMutex};
+          for (auto it : PIDs) {
+            ImGui::PushID(it.first);
```

Comment:
```suggestion
          for (const auto& it : PIDs) {
            ImGui::PushID(it.first);
```

Meant to catch this on my first pass, but this would avoid churning allocation copies just a little bit, since the maps wouldn't be copied

Reply:
Resolved.